arangod/Scheduler/SupervisedScheduler.cpp

```diff
@@ -454,37 +451,54 @@ void SupervisedScheduler::sortoutLongRunningThreads() {
   }
 }
 
+bool SupervisedScheduler::canPullFromQueue(uint64_t queueIndex) const {
+  // This function should ensure the following thread reservation:
+  // 25% reserved for FastLane only
+  // upto 75% of work can go on MedLane and FastLane
+  // uptop 50% of work can go on Slow, Med and FastLane
+  switch (queueIndex) {
+    case 0:
+      // We can always! pull from high priority
+      return true;
+    case 1:
+      // We can work on med if less than 75% of the workers are busy
+      return (_jobsDequeued - _jobsDone) < (_maxNumWorker * 3 / 4);
+    default:
+      // We can work on low if less than 50% of the workers are busy
+      return (_jobsDequeued - _jobsDone) < (_maxNumWorker / 2);
+  }
+}
+
 std::unique_ptr<SupervisedScheduler::WorkItem> SupervisedScheduler::getWork(
     std::shared_ptr<WorkerState>& state) {
   WorkItem* work;
 
   while (!state->_stop) {
     uint64_t triesCount = 0;
     while (triesCount < state->_queueRetryCount) {
```

Comment:
Did you ajust `_queueRetryCount` so that we have more than 3 rounds?

Reply:
No, and I do not think that it is necessary.
The queueIdx `1` has not been used before, only `0` and `2` so I only added the option that `1` actually contains something...